### PR TITLE
Solution: abbreviate-string

### PR DESCRIPTION
**Pull request by AamirAlam**

### CODE REVIEW

File: js-exercises/abbreviate-string/abbreviateString.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// import { isString } from "util";
+
+function abbreviateString(str) {
+    if (typeof str !== 'string'){
+        throw 'Invalid argument! it must be a string';
+    }
+    const allWords = str.split(' ')
```
Comment (tejas-hosamani) [inline js-exercises/abbreviate-string/abbreviateString.js]: The problem that I thought of this approach is, we only need the first and last words. It doesn't make sense to split all the words into an array just to get first and last. All the operation seems redundant to me. 

Hence, my approach was to use `str.indexOf(' ')` to find the first word and `str.lastIndexOf(' ')` to find the last word. 

The `lastIndexOf` method searches for a match from the end of a string, so less operation in theory.

File: js-exercises/abbreviate-string/abbreviateString.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    const firstWord = allWords[0];
+    const lastWord = allWords[allWords.length-1];
```
Comment (tejas-hosamani) [inline js-exercises/abbreviate-string/abbreviateString.js]: Did you consider when the given string has only one word? First and last words would be the same, I believe.

File: js-exercises/abbreviate-string/abbreviateString.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+    return firstWord+" "+lastWord[0].toUpperCase()+"."
```
Comment (tejas-hosamani) [inline js-exercises/abbreviate-string/abbreviateString.js]: Use the modern `es6` feature, `template strings` instead of `es5` method of concatenating strings.

File: js-exercises/abbreviate-string/abbreviateString.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { abbreviateString } from './abbreviateString';
+
+describe('abbreviateString', () => {
+  it('abbreviates the following strings', () => {
+    expect(abbreviateString('Hacktober Fest')).toEqual('Hacktober F.');
+    expect(abbreviateString('Leeroy Fitzgerald Jenkins')).toEqual('Leeroy J.');
+    expect(abbreviateString('Some arbitrary string length here.')).toEqual('Some H.');
```
Comment (tejas-hosamani) [inline js-exercises/abbreviate-string/abbreviateString.test.js:7]: I think if you had written an extra test, you'd have thought of few other edge cases too :)

File: js-exercises/abbreviate-string/abbreviateString.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// import { isString } from "util";
+
+function abbreviateString(str) {
+    if (typeof str !== 'string'){
+        throw 'Invalid argument! it must be a string';
```
Comment (tejas-hosamani) [inline js-exercises/abbreviate-string/abbreviateString.js]: Check out different types of errors you can throw. In this case, `TypeError` seems appropriate to me. Not 100% sure though. 

Something like:
```js
throw new TypeError('Input must be a string');
```

**Review by tejas-hosamani — CHANGES_REQUESTED** [review]
Again, I'd consider making these changes.

Overall, the code is simple, readable, understandable to me. Cheers! 🥂

File: js-exercises/abbreviate-string/abbreviateString.js (hunk 1 above)
Comment (AamirAlam) [reply js-exercises/abbreviate-string/abbreviateString.js]: Yeah make sense

File: js-exercises/abbreviate-string/abbreviateString.js (hunk 2 above)
Comment (AamirAlam) [reply js-exercises/abbreviate-string/abbreviateString.js]: Yes, it will break :(

File: js-exercises/abbreviate-string/abbreviateString.js (hunk 5 above)
Comment (AamirAlam) [reply js-exercises/abbreviate-string/abbreviateString.js]: Yes, the type error looks more relevant, I did not new about that thanks to pointing it out I'll use it now :)
